data/dataset/postgres_example_custom_request_field_dataset.yml

```diff
@@ -0,0 +1,22 @@
+dataset:
+  - fides_key: postgres_example_custom_request_field_dataset
+    data_categories: []
+    description: Postgres example dataset with a custom request field
+    name: Postgrex Example Custom Request Field Dataset
+    collections:
+      - name: dynamic_email_address_config
+        fields:
+          - name: id
+            data_categories: [system.operations]
+            fides_meta:
+              data_type: string
+              primary_key: True
+          - name: email_address
+            data_categories: [system.operations]
+            fides_meta:
+              data_type: string
+          - name: custom_field
+            data_categories: [system.operations]
+            fides_meta:
+              data_type: string
+              custom_request_field: custom_field
```

Comment:
I wonder if it makes more sense for `custom_request_field` to be a boolean? Right now I'm not using its value. I think we have two options here: 
1. Make custom_request_field a boolean. In this case the name of the DSR field is expected to match the name of the field in the dataset
2. Keep the string, which will represent the name of the custom field in the DSR (potentially different than the name of the field in the dataset) , and then map the DSR value to the field value

Reply:
it definitely adds more flexibility, and we can keep it if we see it being useful. In that case I need to add the corresponding logic to map the DSR value to the field value